app/controllers/admin/tour_categories_controller.rb

```diff
@@ -0,0 +1,62 @@
+class Admin::TourCategoriesController < Admin::BaseController
+  before_action :find_tour_category, only: %i(show edit update destroy)
+
+  def index
+    @pagy, @tour_categories = pagy(TourCategory.new_category)
+  end
+
+  def show; end
+
+  def edit; end
+
+  def update
+    if @tour_category.update(tour_category_params)
+      flash[:success] = t("tour_categories.flash_messages.edit_success")
+      redirect_to admin_tour_categories_path
+    else
+      flash[:danger] = t("tour_categories.flash_messages.edit_failure")
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def create
+    @tour_category = TourCategory.new(tour_category_params)
+    if @tour_category.save
+      flash[:success] = t("tour_categories.flash_messages.create_success")
+      redirect_to admin_tour_categories_path
+    else
+      flash[:danger] = t("tour_categories.flash_messages.create_failure")
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
```

Comment:
Giả sử xóa hoặc chỉnh sửa category đã có tour r thì sao em?

Reply:
em đang để dependent là destroy nên nó xóa lun ạ

Reply:
Thế nếu tour đó đang có người đặt thì sao nhỉ :thinking:

Reply:
em sửa thành dependent: :restrict_with_error với in lỗi rùi ạ